tmail-backend/imap-extensions/src/main/java/com/linagora/tmail/imap/TMailMailboxNameConverter.java

```diff
@@ -0,0 +1,227 @@
+package com.linagora.tmail.imap;
+
+import java.util.List;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.Username;
+import org.apache.james.imap.api.display.ModifiedUtf7;
+import org.apache.james.imap.api.process.ImapSession;
+import org.apache.james.imap.main.PathConverter;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.search.MailboxQuery;
+import org.apache.james.mailbox.model.search.PrefixedRegex;
+import org.apache.james.mailbox.model.search.Wildcard;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.google.common.collect.Iterables;
+import com.google.common.escape.Escaper;
+import com.google.common.escape.Escapers;
+import com.linagora.tmail.team.TeamMailbox;
+import com.linagora.tmail.team.TeamMailboxNameSpace;
+
+/**
+ * Fork from org.apache.james.imap.main.PathConverter.Default
+ */
+public class TMailMailboxNameConverter implements PathConverter {
```

Comment:
I am not OK with the level of duplication of this class.

I would rather prefer using `DefaultPathConverter` as a delegate when we are without namespace, `#private` and `#user` while writing custom code to manage `#team-mailboxes`.

This copy means we can have a dangerous delta with James code, meaning any bug fixed upstream won't be fixed here. We shall duplicate code to the minimum.